File: seller-service/src/main/java/com/shopflow/seller/service/SellerService.java
package com.shopflow.seller.service;

import com.shopflow.seller.dto.*;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public interface SellerService {

    SellerResponse registerSeller(UUID userId, SellerRegistrationRequest request);

    SellerResponse getSellerProfile(UUID userId);

    SellerResponse updateStore(UUID userId, StoreUpdateRequest request);

    BigDecimal getEarnings(UUID userId);

    CouponResponse createCoupon(UUID userId, CouponRequest request);

    List<CouponResponse> getCoupons(UUID userId);

    void deleteCoupon(UUID userId, UUID couponId);

    CouponValidationResponse validateCoupon(String code, BigDecimal orderTotal);

    SellerInternalResponse getSellerInternal(UUID sellerId);

    Page<SellerResponse> getAllSellers(Pageable pageable);
    List<SellerResponse> getPendingSellers();
    void approveSeller(UUID sellerId);
    void rejectSeller(UUID sellerId);
    void updateCommission(UUID sellerId, BigDecimal commissionRate);

}

File: shopflow-frontend/src/services/modules/seller.service.js
import axiosInstance from '../api/axiosInstance';

export const sellerService = {
  registerSeller: async (storeData) => {
    return await axiosInstance.post('/sellers/register', storeData);
  },

  getSellerProfile: async () => {
    const response = await axiosInstance.get('/sellers/me');
    return response.data;
  },

  updateStore: async (storeData) => {
    return await axiosInstance.put('/sellers/me/store', storeData);
  },

  getEarnings: async () => {
    const response = await axiosInstance.get('/sellers/me/earnings');
    return response.data;
  },

  createCoupon: async (couponData) => {
    return await axiosInstance.post('/sellers/me/coupons', couponData);
  },

  getCoupons: async () => {
    const response = await axiosInstance.get('/sellers/me/coupons');
    return response.data;
  },

  deleteCoupon: async (id) => {
    return await axiosInstance.delete(`/sellers/me/coupons/${id}`);
  }
};
